src/main/java/spoon/reflect/path/CtRole.java

```diff
@@ -197,11 +109,18 @@ public static CtRole fromName(String name) {
 	 */
 	public String getCamelCaseName() {
 		String s = name().toLowerCase();
-		int i = s.indexOf("_");
-		if (i != -1) {
-			s = s.substring(0, i) + Character.toUpperCase(s.charAt(i + 1)) + s.substring(i + 2);
+		String[] tokens = s.split("_");
+		if (tokens.length == 1) {
+			return s;
+		} else {
+			StringBuffer buffer = new StringBuffer(tokens[0]);
```

Comment:
I think you can use a StringBuilder here: https://stackoverflow.com/questions/355089/difference-between-stringbuilder-and-stringbuffer